x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as React from 'react';
+import ReactDOM from 'react-dom';
+import { AppMountContext, AppMountParameters } from 'kibana/public';
+import { I18nProvider, FormattedMessage } from '@kbn/i18n/react';
+
+/**
+ * This module will be loaded asynchronously to reduce the bundle size of your plugin's main bundle.
+ */
+export function renderApp(appMountContext: AppMountContext, { element }: AppMountParameters) {
+  appMountContext.core.http.get('/api/endpoint/hello-world');
+
+  ReactDOM.render(<AppRoot />, element);
```

Comment:
I think, for internationalization in a NP app, you want:

```
export function renderApp(appMountContext: AppMountContext, { element }: AppMountParameters) {
  appMountContext.core.http.get('/api/endpoint/hello-world');

  ReactDOM.render(
    <appMountContext.core.i18n.Context>
      <AppRoot />
    </appMountContext.core.i18n.Context>,
    element
  );

  return function() {
    ReactDOM.unmountComponentAtNode(element);
  };
}

const AppRoot = React.memo(function Root() {
  return (
    <h1 data-test-subj="welcomeTitle">
      <FormattedMessage id="xpack.endpoint.welcomeTitle" defaultMessage="Hello World" />
    </h1>
  );
});
```

cc @Bamieh -- is that correct?  https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#internationalization recommends `I18nContext` but it's showing the legacy way of importing.  This also seems to contradict the README which shows I18nProvider still... and the Guideline doesn't mention either term.  

Maybe hold off on changing, @oatkiller, until we get confirmation on the right way to do it.

Reply:
@stacey-gammon thanks for the help.